web-ui/src/main/resources/catalog/components/edit/multilingualfield/MultilingualFieldDirective.js

```diff
@@ -74,9 +74,15 @@
           scope.languages = angular.fromJson(attrs.gnMultilingualField);
           var mainLanguage = scope.mainLanguage;
           // Have to map the main language to one of the languages in the inputs
-          if (angular.isDefined(scope.languages[mainLanguage])) {
-            mainLanguage = scope.languages[mainLanguage].substring(1);
-          } else {
+          var langDefined = false;
+          angular.forEach(scope.languages, function(value, key) {
```

Comment:
I believe the logic should be similar to the following in order to be consistent.

https://github.com/geonetwork/core-geonetwork/blob/3895ff84d8612694a75f129a089b44cc9711c47d/schemas/iso19139/src/main/plugin/iso19139/convert/functions.xsl#L152-L162

So it should only apply the special logic if there is a ";" in the text.
And it should assume that the full language code is before the ";" 
I also don't believe there is a need to loop through all the languages to find the correct value.

Reply:
The mainLanguage is fra. But langauges set has the extra "#" in front and it's "#fra". I am looking for where that was set. Maybe we can fix it in metadata not this widget.  

![image](https://user-images.githubusercontent.com/74916635/123327651-a931cd00-d508-11eb-99db-942d8f8e8339.png)

Reply:
The #fra is a pt local language id and it is specific to the metadata. 
i.e. 
<gmd:PT_Locale id="fra">

So for example you could have the following metadata PT_Locale

<gmd:PT_Locale id="fra_id">
<gmd:languageCode>
<gmd:LanguageCode codeList="http://nap.geogratis.gc.ca/metadata/register/napMetadataRegister.xml#IC_116" codeListValue="fra">French; Français</gmd:LanguageCode>
</gmd:languageCode>
<gmd:country>
</gmd:PT_Locale>

In this example the language code would be fra_id and it would be referenced as #fra_id.
It is associated to language code with codelistvalue of fra

So fra is linked to #fra_id

In some cases fra is converted to fre because different iso codes are used (i.e. ISO 639-2B vs ISO 639-2T). I'm not exactly sure why both codes are interchanged within the system? I'm guessing that geonetwork uses  ISO 639-2B and HNAP uses ISO 639-2T)

So fre could also be linked to #fra_id

That is probably how you got fre linked to #fra in your example.

Hopefully that information helps identify the problem.

Reply:
I think I found the issue. This Angular widget "scope.languages[mainLanguage])" is good here. The main language passed from metadata is fra which is taken from the metadata template fra;CAN. However the JSON of all languages is passed as fre: "#fra". The key is fre here. So scope.languages['fra']) will not find anything.

I will investigate in metadata if we can use "fra" as standard instead of fre.

Reply:
I think I found a way to pass the right JSON from metadata to this. What I can see is this widget current system works better than fra which is ISO639_2T standard instead. The only change needed is for translation. Now the en-core.json and fr-core.json has only "fre": "Français", and I will have to add another label "fra": "Français". So this is the only needed change here.

Reply:
I believe geonetwork is using the ISO 639-2B which is why the use fre.  I'm not 100% on this some maybe someone from the community can confirm this?

We cannot simply add "fra" because that will just add more confusion.  And also, there are other languages that have the same issue - not only French.

I believe the problem may be that there should be conversion somewhere to ensure that the ISO 639-2T codes are converted to ISO 639-2B codes.

Reply:
If fre ISO 639-2B is the only choice, then the metadata should not pass anything like fra to this widget at all. What I can see the logic here is the JSON is passed as fre: "#fra". If it finds the mainLanguage fre defined here, it will reasign its value "#fra" without "#" sign back. So it makes the language code inconsistent.

Maybe there is a workaround. So far I found if I force to do "fre":"#fre". The widget will add some "null" fields. Guess there are more issues to be investigated.

Reply:
I found a solution in HNAP. Made a PR https://github.com/metadata101/iso19139.ca.HNAP/pull/223

There is nothing change needed in Geonetwork. I will close this PR.